plugins/woocommerce/src/Blocks/BlockTypes/CheckoutOrderSummaryBlock.php

```diff
@@ -11,4 +11,68 @@ class CheckoutOrderSummaryBlock extends AbstractInnerBlock {
 	 * @var string
 	 */
 	protected $block_name = 'checkout-order-summary-block';
+
+	/**
+	 * Get the contents of the given inner block.
+	 *
+	 * @param string $block_name Name of the order summary inner block.
+	 * @param string $content    The content to search.
+	 * @return array|bool
+	 */
+	private function get_inner_block_content( $block_name, $content ) {
+		if ( preg_match( $this->inner_block_regex( $block_name ), $content, $matches ) ) {
+			return $matches[0];
+		}
+		return false;
+	}
+
+	/**
+	 * Get the regex that will return an inner block.
+	 *
+	 * @param string $block_name Name of the order summary inner block.
+	 * @return string Regex pattern.
+	 */
+	private function inner_block_regex( $block_name ) {
+		return '/<div data-block-name="woocommerce\/checkout-order-summary-' . $block_name . '-block"(.+?)>(.*?)<\/div>/si';
+	}
+
+	/**
+	 * Render the Checkout Order Summary block.
+	 *
+	 * @param array  $attributes Block attributes.
+	 * @param string $content    Block content.
+	 * @param object $block      Block object.
+	 * @return string Rendered block.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		// The order-summary-totals block was introduced as a new parent block for the totals
+		// (subtotal, discount, fees, shipping and taxes) blocks.
+		$regex_for_checkout_order_summary_totals = '/<div data-block-name="woocommerce\/checkout-order-summary-totals-block"(.+?)>/';
+		$order_summary_totals_content            = '<div data-block-name="woocommerce/checkout-order-summary-totals-block" class="wp-block-woocommerce-checkout-order-summary-totals-block">';
+
+		// We want to move these blocks inside a parent 'totals' block.
+		$totals_inner_blocks = array( 'subtotal', 'discount', 'fee', 'shipping', 'taxes' );
+
+		if ( ! preg_match( $regex_for_checkout_order_summary_totals, $content ) ) {
+			foreach ( $totals_inner_blocks as $key => $block_name ) {
+				$inner_block_content = $this->get_inner_block_content( $block_name, $content );
+
+				if ( $inner_block_content ) {
+					$order_summary_totals_content .= "\n" . $inner_block_content;
+
+					// The last block is replaced with the totals block.
+					if ( count( $totals_inner_blocks ) - 1 === $key ) {
+						$order_summary_totals_content .= '</div>';
+						$content                       = preg_replace( $this->inner_block_regex( $block_name ), $order_summary_totals_content, $content );
```

Comment:
I'm not sure why we have to do regex replacements on render. Could you explain why this is required? I only ask because it seems expensive to do it on render rather than defining the structure somewhere else.

Reply:
Thanks for explaining, I just wanted to understand. Happy with this approach then if it's precedented.